contracts/src/Summa.sol

```diff
@@ -9,163 +9,128 @@ import "@openzeppelin/contracts/access/Ownable.sol";
-    mapping(bytes32 => bool) public confirmedAddress;
-
-    //A mapping of asset type hash to the balance retriever smart contract for that type of asset
-    mapping(bytes32 => IBalanceRetriever) public balanceRetriever;
-    //A mapping of address type hash to the address ownership verifier smart contract for that type of address
-    mapping(bytes32 => IAddressOwnershipVerifier) public cexAddressVerifier;
-
     /**
-     * @dev Struct representing an address owned by the CEX
-     * @param addressType The type of the address by chain (e.g., EVM, BTC)
-     * @param cexAddress The address owned by the CEX
-     * @param ownershipProof The data needed to verify the address ownership (e.g., in case of EVM, a signature and a message signed by the address, encoded as bytes)
+     * @dev Struct representing an address ownership proof submitted by the CEX
+     * @param cexAddress The address owned by the CEX (submitted as a string, as it can be a non-EVM address)
+     * @param chain The name of the chain name where the address belongs (e.g., ETH, BTC)
+     * @param signature The signature of the message signed by the address public key
+     * @param message The message signed by the address public key
      */
-    struct OwnedAddress {
-        bytes32 addressType;
-        bytes cexAddress;
-        bytes ownershipProof;
+    struct AddressOwnershipProof {
+        string cexAddress;
+        string chain;
+        bytes signature;
+        bytes message;
     }
 
     /**
      * @dev Struct representing an asset owned by the CEX
-     * @param assetType The type of the asset (e.g., a hash of the asset kind, like keccak256("ETH"), keccak256("BTC", keccak256("ERC20")
-     * @param amountToProve The amount of the asset that the CEX wants to prove to own
-     * @param addresses CEX addresses that control the @param asset
-     * @param balanceRetrieverArgs Additional arguments needed to get the balance (e.g., in case of ERC20, the address of the ERC20 contract)
+     * @param assetName The name of the asset
+     * @param chain The name of the chain name where the asset lives (e.g., ETH, BTC)
      */
-    struct OwnedAsset {
-        bytes32 assetType;
-        uint256 amountToProve;
-        bytes[] addresses;
-        bytes balanceRetrieverArgs;
+    struct Asset {
+        string assetName;
+        string chain;
+        uint256 amount;
     }
 
-    event AddressVerifierSet(bytes32 indexed addressType, address verifier);
-    event BalanceRetrieverSet(
-        bytes32 indexed assetType,
-        address balanceRetriever
+    /**
+     * @dev Zero-knowledge proof and its inputs
+     * @param proof zero-knowledge proof
+     * @param publicInputs proof inputs
+     */
+    struct ZKProof {
+        bytes proof;
+        uint256[] publicInputs;
+    }
+
+    IVerifier private immutable verifier;
+
+    //All address ownership proofs submitted by the CEX
+    AddressOwnershipProof[] public addressOwnershipProofs;
```

Comment:
Both the address ownership proofs and the proofs of solvency are:
1. Written to the smart contracts => `addressOwnershipProofs` and `proofOfSolvency`
2. Emitted in events `ExchangeAddressesSubmitted` and `ProofOfSolvencySubmitted`

The goal of the contract is to store (or log) these information such that external actors can challenge it if malicious. 

I'm wondering if you see a difference between writing these to the smart contract and emitting them through events or is it just redundancy.

Reply:
- I think that the process of signing proof might possibly added in the future but is not in plan now. Maybe it will be added after the discussion with exchanges 
- If we don't need to store the proofOfSolvency I think this can be safely removed.